### PR TITLE
Json parser

### PR DESCRIPTION
**Pull request by garunagiri**
Tested this parser with custom JSON output. Error conditions were written by looking at the way katz.py logs errors, but not an exact match. If they should be different I can change them.

### CODE REVIEW

File: app/parsers/json.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+from app.objects.c_relationship import Relationship
+from plugins.stockpile.app.parsers.base_parser import BaseParser
+import json
```
Comment (wbooth) [inline app/parsers/json.py]: https://www.python.org/dev/peps/pep-0008/#id23

File: app/parsers/json.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+class Parser(BaseParser):
+
+    def parse(self, blob):
+        relationships = []
+        try:
+            json_obj = json.loads(blob)
+        except Exception as error:
+            self.log.warning("Output not JSON, use a different parser")
+            return []
```
Comment (wbooth) [inline app/parsers/json.py]: move to private function

File: app/parsers/json.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                source = self.set_value(mp.source, match, self.used_facts)
+                target = self.set_value(mp.target, match, self.used_facts)
+                relationships.append(
+                    Relationship(source=(mp.source, source),
+                                 edge=mp.edge,
+                                 target=(mp.target, target))
+                )
+        return relationships
+
+    def get_vals_from_json(self, json_obj, key):
```
Comment (wbooth) [inline app/parsers/json.py]: rename and make private

Comment (wbooth) [reply app/parsers/json.py]: rename or add docstring describing its behavior, it gets all values for key recursively or it gets first value, etc.

**Comment by garunagiri** [comment]
@wbooth Are these changes what you were looking for?

File: app/parsers/json.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        self.used_facts = parser_info['used_facts']
+        self.log = Logger('parsing_svc')
+
+    def parse(self, blob):
+        relationships = []
+        json_output = self._load_json(blob)
+        if json_output is not None:
+            for mp in self.mappers:
+                if "json_key" not in dir(mp):
+                    self.log.warning("JSON Parser not given a 'json_key', not parsing")
```
Comment (wbooth) [inline app/parsers/json.py]: use single quotes

Comment (garunagiri) [reply app/parsers/json.py]: Changed

**Comment by wbooth** [comment]
@garunagiri do you have an example of this in practice?

**Pull request merged**